Shared/alerts/AlertConditionsController.cpp

```diff
@@ -402,18 +406,30 @@ bool AlertConditionsController::addAttributeEqualsAlert(const QString& condition
     return false;
   }
 
-  AlertTarget* target = new FixedValueAlertTarget(targetValue, this);
-
-  GraphicsOverlay* sourceOverlay = graphicsOverlayFromName(sourceFeedName);
-  if (!sourceOverlay)
+  GraphicsOverlay* sourceGraphicsOverlay = graphicsOverlayFromName(sourceFeedName);
+  DynamicEntityLayer* sourceDynamicEntityLayer = nullptr;
+  if (!sourceGraphicsOverlay)
+  {
+    sourceDynamicEntityLayer = dynamicEntityLayerFromName(sourceFeedName);
+  }
+  if (!sourceGraphicsOverlay && !sourceDynamicEntityLayer)
   {
     emit toolErrorOccurred(QStringLiteral("Failed to create Condition"), QString("Could not find source feed: %1").arg(sourceFeedName));
     return false;
   }
 
+  AlertTarget* target = new FixedValueAlertTarget(targetValue, this);
   AttributeEqualsAlertCondition* condition = new AttributeEqualsAlertCondition(level, conditionName, attributeName, this);
   connect(condition, &AttributeEqualsAlertCondition::newConditionData, this, &AlertConditionsController::handleNewAlertConditionData);
-  condition->init(sourceOverlay, sourceFeedName, target, targetValue.toString());
+
+  if (sourceGraphicsOverlay)
+  {
+    condition->init(sourceGraphicsOverlay, sourceFeedName, target, targetValue.toString());
+  }
+  else
+  {
+    condition->init(sourceDynamicEntityLayer, sourceFeedName, target, targetValue.toString());
+  }
```

Comment:
This approach was setup to avoid initializing anything unnecessarily but also to minimize repeating any code. Since these layers don't share an ancestor, I have switched to a templated function to avoid duplicating the same code that is necessary later in the method. Please let me know what you think.